libraries/standard/http/src/http_client.c

```diff
@@ -132,10 +151,171 @@ static HTTPStatus_t _addHeader( HTTPRequestHeaders_t * pRequestHeaders,
 
 /*-----------------------------------------------------------*/
 
+static HTTPStatus_t _writeRequestLine( HTTPRequestHeaders_t * pRequestHeaders,
+                                       const char * pMethod,
+                                       size_t methodLen,
+                                       const char * pPath,
+                                       size_t pathLen )
+{
+    HTTPStatus_t returnStatus = HTTP_SUCCESS;
+    uint8_t * pBufferCur = pRequestHeaders->pBuffer;
+    size_t toAddLen = methodLen +                 \
+                      SPACE_CHARACTER_LEN +       \
+                      SPACE_CHARACTER_LEN +       \
+                      HTTP_PROTOCOL_VERSION_LEN + \
+                      HTTP_HEADER_LINE_SEPARATOR_LEN;
+
+    assert( pRequestHeaders != NULL );
+    assert( pRequestHeaders->pBuffer != NULL );
+    assert( pMethod != NULL );
+    assert( methodLen != 0u );
+
+    toAddLen += ( pPath == NULL || pathLen == 0 ) ? HTTP_EMPTY_PATH_LEN : pathLen;
+
+    if( ( toAddLen + pRequestHeaders->headersLen ) > pRequestHeaders->bufferLen )
+    {
+        returnStatus = HTTP_INSUFFICIENT_MEMORY;
+    }
+
+    if( returnStatus == HTTP_SUCCESS )
+    {
+        /* Write "<METHOD> <PATH> HTTP/1.1\r\n" to start the HTTP header. */
+        memcpy( pBufferCur, pMethod, methodLen );
```

Comment:
I suppose for consistency we will want to use snprintf for this as we had in other places. 

But I am OK with memcpy. There are tradeoffs to snprintf (such as encoding errors), while memcpy will copy as it is.